### PR TITLE
Basic readme and ci setup for compiling chapter code

### PR DESCRIPTION
**Pull request by i509VCB**
The first commit of the book for the 0.30 cycle!

This is commit is preparation for the new book (since we will effectively nuke the old book) to be written.

What does this commit introduce?
- CI infrastructure to test if the book actually builds (since it can)
- Allow chapters to have their code compile checked.
  - We can do this since mdbook allows us to use code snippets from other files using `{{#rustdoc_include <FILE>:<ANCHOR>}}` templates inside code blocks.
- Also added a simple readme. `book` isn't all that descriptive.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,37 @@
...
+
+To build the book, run the following:
+
+```
+$ mdbook build
+```
+
+The output will be in the `book` subdirectory. The book can then be previewed in your web browser by opening the `index.html` file in the book subdirectory.
+
+The book is also tested to ensure the source code in the book is correct. To run the tests, run the following:
```
Comment (i509VCB) [inline README.md]: Actually this isn't applicable anymore...

Comment (elinorbgr) [reply README.md]: Why not use `mdbook test` btw?

File: Cargo.lock [hunk 2]
```diff
@@ -0,0 +1,403 @@
+# This file is automatically @generated by Cargo.
```
Comment (elinorbgr) [inline Cargo.lock]: Did you plan to commit the `Cargo.lock`? If so, why?

Comment (i509VCB) [reply Cargo.lock]: Didn't plan this

File: README.md (hunk 1 above)
Comment (i509VCB) [reply README.md]: Mdbook test does not support 3rd party crates. It's rust stdlib or bust

Comment (elinorbgr) [reply README.md]: :+1:

File: Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+[package]
+name = "smithay-book"
+version = "0.0.1"
+edition = "2021"
```
Comment (elinorbgr) [inline Cargo.toml:4]: Just to be safe, let's add a `publish = false` here

**Pull request merged**